src/value/owned.rs

```diff
@@ -187,52 +197,113 @@ impl PartialEq for Sequence {
 
 impl Eq for Sequence {}
 
+// This collection is broken out into its own type to allow instances of it to be shared with Rc.
+#[derive(Debug)]
+struct Fields {
+    // Key/value pairs in the order they were inserted
+    by_index: Vec<(Symbol, Element)>,
+    // Maps symbols to a list of indexes where values may be found in `by_index` above
+    by_name: HashMap<Symbol, IndexVec>,
+}
+
+impl Fields {
+    /// Gets all of the indexes that contain a value associated with the given field name.
+    fn get_indexes<A: AsSymbolRef>(&self, field_name: A) -> Option<&IndexVec> {
+        match field_name.as_symbol_ref().text() {
+            // If the provided field name symbol has undefined text...
+            None => {
+                // ...then build a cheap, stack-allocated `Symbol` that represents unknown text
+                let symbol = Symbol::unknown_text();
+                // ...and use the unknown text symbol to look up matching field values
+                self.by_name.get(&symbol)
+            }
+            Some(text) => {
+                // Otherwise, look it up by text
+                self.by_name.get(text)
+            }
+        }
```

Comment:
```suggestion
        field_name.as_symbol_ref().text()
            .and_then(|text| self.by_name.get(text))
            .or_else(|| self.by_name.get(&(Symbol::unknown_text())))
```

It looks like something like this will work here. I think it's clearer, this is the kind of logic these combinators were written for.

My suggestion here doesn't preserve the comments or the first unknown then known ordering, which is not ideal. Take or leave as little as you like :)